overlord/snapstate/snapstate.go

```diff
@@ -805,8 +805,8 @@ func FinishRestart(task *state.Task, snapsup *SnapSetup) (err error) {
 // RestartSystem requests a system restart.
 // It considers how the Change the task belongs to is configured
 // (system-restart-immediate) to choose whether request an immediate
-// restart or not.
-func RestartSystem(task *state.Task, rebootInfo *boot.RebootInfo) {
+// restart or not, then it calls FinishTaskWithRestart.
+func RestartSystem(task *state.Task, status state.Status, rebootInfo *boot.RebootInfo) error {
```

Comment:
I think it's still slightly tempting to merge this into FinishTaskWithRestart from a surface pov. But no super strong opinion, OTOH I think if we keep both, we should define FinishTaskWithRestart before this one.

Reply:
I've done this now, I think it simplifies the code, thanks for the suggestion.